### DISCUSSION TITLE
How to make a rectangle with rounded corners

### DISCUSSION THREAD

**Discussion by Paul8043** [CadQuery/cadquery#1297] [General]
I want to create a rectangle with rounded corners (see line 3) for doing later some extrudes (see line 4). The whole thing runs fine for 3D-version (second line), but not for the 2D-version (third line).

```
import cadquery as cq
#p3d = cq.Workplane("XY" ).box(10, 10, 10).edges().fillet(1)
p2d = cq.Workplane("XY" ).rect(10, 10).vertices().fillet(1)
#part = cq.Workplane("XY" ).rect(10, 10).vertices().fillet(1).extrude(2)
```

The Cheatsheet  says there is also a fillet() for 2D, but it is listed in the Sketching section. It is not clear, when to sketch, and when not to sketch? Can you explain the very nature of a sketch and how it should be defined and used? I have read the manual section, but I cannot put the puzzle together. It looks that a sketch is something like a macro which holds several operations together. It is defined without a workplane context. But again, it is unclear, how the sketch is applied later on to a face or whatever.

Can you explain how
reset()
clean()
close()
finalize()
should be used?

Nothing helps more than a running example.

**Comment by jmwright** [comment]
> It is not clear, when to sketch, and when not to sketch?

Sketch is designed as a replacement for the old style 2D sketching, which does not support things like fillets, cuts and intersects. I still use the old style of 2D for simple things (normal rects, circles, etc), but Sketch must be used for those advanced 2D features.

Here is a rect with filleted corners via Sketch.

```python
result = (
   cq.Sketch()
   .rect(4, 4)
   .vertices()
   .fillet(0.25)
)
```
